pyk/src/pyk/testing/_kompiler.py

```diff
@@ -300,6 +304,45 @@ def kcfg_explore(
         yield KCFGExplore(cterm_symbolic, kcfg_semantics=semantics)
 
 
+class KoreClientFactoryTest(KoreServerTest):
+    KOMPILE_BACKEND: ClassVar = 'haskell'
+    CLIENT_TIMEOUT: ClassVar = 1000
+
+    # definition_dir should ideally depend on server_type to avoid triggering kompilation,
+    # but it can't due to method signature in superclass.
+    # In the parameter list, always put `definition_dir` after `kore_client`.
+    @pytest.fixture
+    def create_kore_client(self, _kore_server: KoreServer, bug_report: BugReport) -> Callable[[], KoreClient]:
```

Comment:
Since tests do not seem to consume this fixture directly, I'd just move this into `ParallelTest` as a private helper method.

Reply:
Fixed.